src/vm/i386/umthunkstub.S

```diff
@@ -173,25 +173,20 @@ LOCAL_LABEL(UMThunkStub_CopyStackArgs):
 LOCAL_LABEL(InitCopyStack):
     push    ecx
     push    edx
-    mov     eax, 0  // Offset in edi
-    mov     ebx, 0  // Offset in esi
-LOCAL_LABEL(CopyStack):
     mov     edx, dword ptr [ebp - UMThunkStub_UMENTRYTHUNK_OFFSET]
     mov     edx, dword ptr [edx + UMEntryThunk__m_pUMThunkMarshInfo]
-    mov     ecx, dword ptr [edx + UMThunkMarshInfo__m_ecxArgOffset]
-    cmp     ecx, ebx
+    mov     ebx, [edx + UMThunkMarshInfo__m_cbActualArgSize]
```

Comment:
The CF is set by the add ebx, -4 before the conditional jump. But I have just realized the `jnc` is wrong, since adding -4, which is 0xfffffffc to 0 doesn't set the CF, but rather clears it. And it is set in all other cases (4 + 0xfffffffc sets it, 8 + 0xfffffffc sets it too, etc).
So the polarity should be reverse, it means `jc`.